provider/godaddy/godaddy_test.go

```diff
@@ -370,7 +371,76 @@ func TestGoDaddyChange(t *testing.T) {
 		},
 	}).Return(nil, nil).Once()
 
+	// Delete entry
+	client.On("Delete", "/v1/domains/example.net/records/A/godaddy").Return(nil, nil).Once()
+
 	assert.NoError(provider.ApplyChanges(context.TODO(), &changes))
 
 	client.AssertExpectations(t)
 }
+
+var (
+	status404 string = "404"
```

Comment:
no need to reinvent the wheel, you can use the status codes and information from http/status pkg, see https://golang.org/src/net/http/status.go

Reply:
As described in the  [GoDaddy Domains API](https://developer.godaddy.com/doc/endpoint/domains), section Error & ErrorField, error code are not http status.

<span class="model-box"><span class="model"><span class=""><span class="inner-object">
code* | string($constant)Short identifier for the error, suitable for indicating the specific error within client code
-- | --
fields | [...]
message | stringHuman-readable, English description of the error

</span></span></span></span>

To avoid confusion, I have changed the structure and variables name